src/EppoClient.php

```diff
@@ -141,202 +120,164 @@ public static function getInstance(): EppoClient
     {
-        try {
-            $assignmentVariation = $this->getAssignmentVariation($subjectKey, $flagKey, $subjectAttributes, self::VARIANT_TYPE_JSON);
-            return $assignmentVariation ? $assignmentVariation->typedValue : null;
-        } catch (Exception $exception) {
-            return $this->handleException($exception);
-        }
+        return $this->getTypedAssignment(VariationType::INTEGER, $flagKey, $subjectKey, $subjectAttributes, $defaultValue);
     }
 
     /**
-     * Gets the assigned JSON variation, represented as JSON string, for the given subject and experiment.
+     * Gets the assigned JSON variation, as parsed by PHP's json_decode, for the given subject and experiment.
      * If there is an issue retrieving the variation or the retrieved variation is not valid JSON, null wil be returned.
      *
-     * @return string|null the parsed variation JSON as a string
+     * @param string $flagKey
+     * @param string $subjectKey
+     * @param array $subjectAttributes
+     * @param array|null $defaultValue
+     * @return array|null the parsed variation JSON
      *
-     * @throws HttpRequestException
-     * @throws GuzzleException
-     * @throws InvalidApiKeyException
-     * @throws InvalidArgumentException
+     * @throws ClientExceptionInterface
      * @throws SimpleCacheInvalidArgumentException
      */
-    public function getJSONStringAssignment(string $subjectKey, string $flagKey, array $subjectAttributes = []): ?string
+    public function getJSONAssignment(string $flagKey, string $subjectKey, array $subjectAttributes = [], array $defaultValue = null): ?array
     {
-        try {
-            $parsedJsonValue = $this->getParsedJSONAssignment($subjectKey, $flagKey, $subjectAttributes);
-            return isset($parsedJsonValue) ? json_encode($parsedJsonValue) : null;
-        } catch (Exception $exception) {
-            return $this->handleException($exception);
-        }
+        return $this->getTypedAssignment(VariationType::JSON, $flagKey, $subjectKey, $subjectAttributes, $defaultValue);
     }
 
     /**
      * Gets the legacy, string-only assignment for the given subject and experiment.
      * If there is an issue retrieving the variation, null wil be returned.
      *
-     * @throws HttpRequestException
-     * @throws GuzzleException
-     * @throws InvalidApiKeyException
-     * @throws InvalidArgumentException
-     * @throws SimpleCacheInvalidArgumentException
+     * @throws SimpleCacheInvalidArgumentException|ClientExceptionInterface
      * @deprecated in favor of the typed get<type>Assignment methods
      *
      */
-    public function getAssignment(string $subjectKey, string $flagKey, array $subjectAttributes = []): ?string
+    public function getAssignment(string $flagKey, string $subjectKey, array $subjectAttributes = [], ?string $defaultValue = null): ?string
```

Comment:
🪓  We are removing the untyped `getAssignment()`s from SDKs, requiring a type to always be specified (along with a default value).

Reply:
great! consider it 🪓 'd